velox/functions/sparksql/tests/ArrayMaxTest.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <type/Timestamp.h>
+#include <limits>
+#include <optional>
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class ArrayMaxTest : public SparkFunctionBaseTest {
+ protected:
+  template <typename T>
+  std::optional<T> arrayMax(const std::vector<std::optional<T>>& input) {
+    std::vector<std::vector<std::optional<T>>> vec = {input};
+    auto row = makeRowVector({makeNullableArrayVector(vec)});
+    return evaluateOnce<T>("array_max(C0)", row);
+  }
+};
+
+TEST_F(ArrayMaxTest, boolean) {
+  EXPECT_EQ(arrayMax<bool>({true, false}), true);
+  EXPECT_EQ(arrayMax<bool>({true}), true);
+  EXPECT_EQ(arrayMax<bool>({false}), false);
+  EXPECT_EQ(arrayMax<bool>({}), std::nullopt);
+  EXPECT_EQ(arrayMax<bool>({true, false, true, std::nullopt}), true);
+  EXPECT_EQ(arrayMax<bool>({std::nullopt, true, false, true}), true);
+  EXPECT_EQ(arrayMax<bool>({false, false, false}), false);
+  EXPECT_EQ(arrayMax<bool>({true, true, true}), true);
+}
+
+TEST_F(ArrayMaxTest, varchar) {
+  EXPECT_EQ(arrayMax<StringView>({"red"_sv, "blue"_sv}), "red"_sv);
+  EXPECT_EQ(
+      arrayMax<StringView>({std::nullopt, "blue"_sv, "yellow"_sv, "orange"_sv}),
+      "yellow"_sv);
+  EXPECT_EQ(arrayMax<StringView>({}), std::nullopt);
+  EXPECT_EQ(arrayMax<StringView>({std::nullopt}), std::nullopt);
+}
+
+// Test non-inlined (> 12 length) nullable strings.
+TEST_F(ArrayMaxTest, longVarchar) {
+  EXPECT_EQ(
+      arrayMax<StringView>(
```

Comment:
StringView doesn't own the string, hence, it is incorrect to return StringView from a function and let the backing string be destroyed.

Perhaps, add template specialization for arrayMax<std::string> to take and return std::strings.

Reply:
Fixed.